e2e_test.go

```
@@ -0,0 +1,286 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"os"
+	"path/filepath"
+	"reflect"
+)
+
+var (
+	storagePath   string
```

ah! here's a good reason to use Ginkgo over go's testing package. Ginkgo has a 'BeforeSuite' function which runs a setup (as well as AfterSuite for teardown) before (and after, respectively) any/all tests begin.

To avoid using a global var here, we could simply attempt to start the server and ignore error as long as the error string matches "address already in use"